### DISCUSSION TITLE
dotnet-codegen not found

### DISCUSSION THREAD

**Discussion by kevinrjones** [louthy/language-ext#996] [General]
Hi,

I have a project that I'm building an a mac (M1) and I'm trying to use language-ext with the code gen.

I'm getting this error

`  CodeGeneration.Roslyn.BuildTime.targets(73, 5): [CGR1001] CodeGeneration.Roslyn.Tool (dotnet-codegen) is not available, code generation won't run. Please check https://github.com/AArnott/CodeGeneration.Roslyn for usage instructions.
`
my csproj looks like this:

`
<Project Sdk="Microsoft.NET.Sdk">

    <PropertyGroup>
        <TargetFramework>net6.0</TargetFramework>
        <LangVersion>10</LangVersion>
    </PropertyGroup>

    <ItemGroup>
        <PackageReference Include="CodeGeneration.Roslyn.BuildTime" Version="0.6.1">
          <PrivateAssets>all</PrivateAssets>
          <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
        </PackageReference>
        <PackageReference Include="CSharpFunctionalExtensions" Version="2.28.0" />
        <PackageReference Include="LanguageExt.CodeGen" Version="4.0.3" />
       
        <PackageReference Include="LanguageExt.Core" Version="4.0.3" />
        <DotNetCliToolReference Include="dotnet-codegen" Version="0.7.63" />

    </ItemGroup>

</Project>
`

Not sure what I'm missing here or even if I'm supposed to be using  Roslyn code generation

I've also tried changing the 'CodeGeneration.Rosylyn.Buildtime' version to '0.7.5-alpha' as I thought I read somewhere about version upgrades on .Net?

What should I be doing to get this working?

Thanks

**Comment by kevinrjones** [comment]
To answer my own question. I think to get this to work I need to install the 2.1 dotnet core SDK, unfortunately there's no ARM version of this for the Mac so I'm guessing it's not going to work

**Reply by harrhp** [reply]
Have you tried this https://github.com/AArnott/CodeGeneration.Roslyn/issues/180#issuecomment-562895670 ?
